packages/mobile/src/invite/JoinCelo.tsx

```diff
@@ -133,7 +131,10 @@ export class JoinCelo extends React.Component<Props, State> {
   }
 
   nextScreen = () => {
-    const nextScreen = this.props.pincodeSet ? Screens.EnterInviteCode : Screens.Pincode
+    const nextScreen =
+      this.props.pincodeType === PincodeType.Unset
```

Comment:
Not completely sure, but I think here you have an edge case when the user does the onboarding flow and then disables the security of their phone (think likely to happen with people that only set up the PIN to use the app). It probably won't explicitly fail here but when they try to send a transaction `getPinFromKeystore` will just throw an error.

So I'd suggest checking every time that ConfirmDeviceCredentials.isDeviceSecure(), specially when sending a transaction.

Reply:
Did you mean this as a general point or something specific to this line? Because it seems unrelated to the JoinCelo page.

In the case where the user removes phone security after onboarding with it, throwing an error in `getPinFromKeystore` seems like the right thing to do. Is there an alternative you'd suggest?

Reply:
As a general point I meant.

Yeah, getting `getPinFromKeystore ` to throw an error is the right call but I think it should be catched so the user can get redirected to the education screen again (maybe with a slightly different copy).

Reply:
Currently, if `getPinFromKeystore` throws an error (of any kind), then `getConnectedUnlockedAccount` will throw an `IncorrectPin` error. I agree it would be good to differentiate incorrect pin from missing phone auth but IIUC it would require modifying everywhere that `getConnectedUnlockedAccount` is used to handle that error case. Not too bad but still a bit out of scope for this PR I think. WDYT?

Reply:
Agreed, opened #1090!